### ISSUE TITLE
Support for Apple silicon natively

### ISSUE THREAD

**Issue by DanielThomas** [gradle/gradle#15509]
I'm testing Gradle on a M1 Mac Mini and everything looks good so far, however Gradle is falling back to lowest common denominator defaults due to lack of JNI bindings for Apple's new processors. For instance:

- The terminal is not detected and the UI falls back to plain output
- File watching is not supported
```
Couldn't create watch service, not tracking changes between builds
net.rubygrapefruit.platform.NativeIntegrationUnavailableException: Native integration OsxFileEventFunctions is not supported for Mac OS X aarch64.
	at net.rubygrapefruit.platform.internal.Platform.get(Platform.java:114)
	at net.rubygrapefruit.platform.Native.get(Native.java:88)
...
```

### Expected Behavior
All native features supported on macOS should work on Apple Silicon.

### Current Behavior
Those features do not work.

### Context
macOS is a supported operating system and needs the native components compiled for `aarch64` and the architecture detected as a supported macOS architecture.

**Comment by lptr** [comment]
To clarify: this is _without_ Rosetta2 installed, correct? I just run some builds _with_ Rosetta2, and both the terminal and file system watching works well:

https://ge.gradle.org/s/2mgbi57cbs4c2/console-log#L4

**Comment by DanielThomas** [comment]
Yes, sorry - I should have been clearer. Intel JVMs of course work fine under Rosetta 2, this is with the Zulu OpenJDK ARMv8 builds for macOS: https://www.azul.com/downloads/zulu-community/?package=jdk

**Comment by lptr** [comment]
Alright. So it's not like people with M1 Macs are blocked or can't use the features, it's that they need to use Rosetta2, right? Just trying to assess the urgency of the situation.

My understanding is that with Rosetta2 you can work unimpeded, but with some loss of performance. OTOH, it seems to require some significant investment on our side to get things working natively. I'll do some more research to see exactly how much work it would be to ship 7.0 with M1 natively supported, but I don't want to commit to shipping it right now.

**Comment by lptr** [comment]
`native-platform` for Apple silicon is taking shape: https://github.com/gradle/native-platform/pull/265

**Comment by DanielThomas** [comment]
That's awesome @lptr! To your earlier message - right, an Intel JVM would work okay, but the performance difference between using an Intel and ARM JVM is significant. Like I said on Slack, we'll recommend folks hold off until Docker, IntelliJ, etc. all have final releases, but feels like that'll happen within a month or two.

**Comment by lptr** [comment]
Looks like it will work. Here's a build scan with a Zulu ARM JDK having file system watching enabled and rich console output: https://scans.gradle.com/s/qb3kjqpnqescs#switches

**Comment by DanielThomas** [comment]
Nice one! Glad it turned out to not be too difficult. Appreciate you picking this up!

**Comment by Smagold** [comment]
I'm having issues using Gradle on my M1 processor:
`Could not create Jansi native library '/usr/local/opt/gradle/native/jansi/1.17.1/osx/libjansi.jnilib'.`

**Comment by wolfs** [comment]
@Smagold Is it possible that Gradle cannot write to that directory? How did you install Gradle?

**Comment by Smagold** [comment]
> @Smagold Is it possible that Gradle cannot write to that directory? How did you install Gradle?

Thank you for the tip. Fixed permissions helped!

**Comment by tyates-indeed** [comment]
Are there plans to backport this fix to Gradle 6.x?

**Comment by lptr** [comment]
It is available in Gradle 6.9: https://docs.gradle.org/6.9/release-notes.html#native-support-for-apple-silicon
